mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,407 @@
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanKills() + toAdd;
+
+        rec.setClanKills(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_KILLS);
+
+        this.clanKills = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds deaths to a gg.solarmc.loader.clans.Clan
+     * @param transaction the tx
+     * @param toAdd amount of deaths to add
+     * @return accurate count of deaths post transaction
+     */
+    public Integer addDeaths(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanDeaths() + toAdd;
+
+        rec.setClanDeaths(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_DEATHS);
+
+        this.clanDeaths = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds assists to a gg.solarmc.loader.clans.Clan.
+     * @param transaction tx
+     * @param toAdd the amount you want to add
+     * @return Accurate assists after transaction
+     */
+    public Integer addAssists(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanAssists() + toAdd;
+
+        rec.setClanAssists(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_ASSISTS);
+
+        this.clanAssists = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Returns the accurate allied clan to this object.
+     * @param transaction
+     * @return
+     */
+    public Optional<Clan> getAlliedClan(Transaction transaction) {
+        if (this.alliedClan == null) return Optional.empty();
+
+        return Optional.of(manager.getClan(transaction,alliedClan));
+    }
+
+    /**
+     * Gets all cached clan members
+     * @param transaction
+     * @return
+     */
+    public Set<ClanMember> getClanMembers(Transaction transaction) {
+        return transaction.getProperty(DSLContext.class)
+                .select(CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .from(CLANS_CLAN_MEMBERSHIP)
+                .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(this.clanID))
+                .fetchSet((rec) -> new ClanMember(this.clanID,rec.value1(),manager));
+    }
+
+    /**
+     * Adds a member to the clan. Note that this does not have a built in limit, API implementer needs to implement
+     * limit if we desire a limit on members
+     *
+     * Note to API implementer: This handles nothing regarding invites. If you want an invite system, it needs to come
+     * before this method.
+     *
+     * @param transaction The tx
+     * @param receiver gg.solarmc.loader.clans.ClanDataObject to add.
+     * @return Accurate set of ClanMembers post transaction
+     * @throws IllegalStateException if member is already a member of another clan
+     * @throws IllegalArgumentException if member is the leader of the clan
+     */
+    public Set<ClanMember> addClanMember(Transaction transaction, ClanDataObject receiver) {
```

Comment:
We should declare SolarPlayer parameters rather than ClanDataObjects. It makes things cleaner for API users.
```suggestion
    public Set<ClanMember> addClanMember(Transaction transaction, SolarPlayer receiver) {
```

Reply:
Finished!